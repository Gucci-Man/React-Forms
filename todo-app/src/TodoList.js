import React, {useState} from 'react';
import Todo from './Todo';
//import NewTodoForm from './NewTodoForm';
import { v4 as uuid } from "uuid";

const TodoList = () => {
    const [todos, setTodo] = useState([])
    const addTodo = (newTodo) => {
        setTodo(todos => [...todos, {...newTodo, id: uuid() }])
    }
    const deleteTodo = (todoId) => {
        setTodo(todos => todos.filter(todo => todo.id !== todoId))
    }
    return (
        <div>
            <h2>Todo List</h2>
            {/* <NewTodoForm addBox={addBox} /> */}
            <div>
                {todos.map(({id, task}) => 
                <Todo
                    id={id} 
                    task={task}
                    key={id}
                    deleteTodo={deleteTodo}
                />)}
            </div>
        </div>
    )
}

export default TodoList;