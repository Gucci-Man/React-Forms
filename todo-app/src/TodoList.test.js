import React from 'react';
import TodoList from './TodoList';
import { render, fireEvent } from "@testing-library/react";

// Smoke test
it("renders without crashing", () => {
  render(<TodoList />);
});

// Snap test
it("matches snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

// Unit tests
it("should add a new todo", () => {
  const { queryByText, getByLabelText } = render(<TodoList />);
  
  expect(queryByText("Reading")).not.toBeInTheDocument();
  
  fireEvent.change(getByLabelText("Task:"), { target: {value: 'Reading' }});
  fireEvent.click(queryByText("Add Todo"));
  
  expect(queryByText("Reading")).toBeInTheDocument();
});

it("should delete a todo", () => {
    const { queryByText, getByLabelText } = render(<TodoList />);

    // Add a box 
    expect(queryByText("Jogging")).not.toBeInTheDocument();
  
    fireEvent.change(getByLabelText("Task:"), { target: {value: 'Jogging' }});
    fireEvent.click(queryByText("Add Todo"));
    
    expect(queryByText("Jogging")).toBeInTheDocument();

    // Delete the box
    fireEvent.click(queryByText("x"));
    expect(queryByText("Jogging")).not.toBeInTheDocument();
});