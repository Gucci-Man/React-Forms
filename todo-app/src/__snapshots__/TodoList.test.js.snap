// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Todo List
    </h2>
    <form>
      <label
        for="task"
      >
        Task: 
      </label>
      <input
        id="task"
        name="task"
        placeholder="Your task"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
